### PR TITLE
DDCE-3492 Fix accessibility issues

### PR DESCRIPTION
**Pull request by mywyau**

### CODE REVIEW

**Comment by mywyau** [comment]
**Latest Report** 
Latest Report

File: app/views/forms/components/repayment_input_checkbox.scala.html [hunk 1]
```diff
@@ -28,23 +28,29 @@
 @field = @{formData("repaymentClaim")}
 
 <div class="govuk-form-group">
-    <fieldset class="govuk-fieldset" aria-describedby="expiry-date-hint">
+    <fieldset class="govuk-fieldset" aria-describedby="@{field.id}">
```
Comment (frikit) [inline app/views/forms/components/repayment_input_checkbox.scala.html:31]: same bug here with checkbox it doesnt remember that it is checked or it is not checking it with there is data inside.

I cant find exact reason here but on http://localhost:9581/manage-tariff-classifications/cases/v2/800000013/liability/edit-details and you have some data in dvr number which is under repayment claim checkbox it doesnt not check it to expand the input to be able to delete even the data is inside, it doesnt not remember or there is not aria-expanded attribute on tag,
there should be aria-expanded="true" but it always false even when there is data inside or there is error

I dont see in old code this in place explicitly but it is working fine there, feel free to play around maybe something else is missing which I could not figure out

File: app/views/includes/govuk_wrapper.scala.html [hunk 2]
```diff
@@ -52,14 +57,14 @@
 
 @govukLayout(
     pageTitle = Some(pageTitle),
-    headBlock = Some(hmrcHead(head)),
-    headerBlock = Some(hmrcStandardHeader(
-        serviceUrl = None,
-        serviceName = Some(messages("service.name")),
-        displayHmrcBanner = false
-    )),
+    headBlock = Some(hmrcInternalHead(head, nonce = CSPNonce.get)),
```
Comment (frikit) [inline app/views/includes/govuk_wrapper.scala.html:60]: Screenshot 2022-12-01 at 12 40 23
not sure if that is expected header but I doubt that :D

File: app/views/includes/govuk_wrapper.scala.html [hunk 3]
```diff
@@ -52,14 +57,14 @@
...
+    headerBlock = Some(hmrcInternalHeader(InternalHeader(
+                    homepageUrl = "/",
```
Comment (frikit) [inline app/views/includes/govuk_wrapper.scala.html:62]: this should be relative url to the app or something, now it does nothing but refresh current page

File: app/views/includes/govuk_wrapper.scala.html [hunk 4]
```diff
@@ -52,14 +57,14 @@
...
+                    serviceName = Some(messages("service.name")),
+                    serviceUrl = "",
```
Comment (frikit) [inline app/views/includes/govuk_wrapper.scala.html:64]: or here this should be relative url to the app or something, now it does nothing but refresh current page

**Review by frikit — CHANGES_REQUESTED** [review]

File: app/views/forms/components/repayment_input_checkbox.scala.html (hunk 1 above)
Comment (mywyau) [reply app/views/forms/components/repayment_input_checkbox.scala.html:31]: reverting to old version not exactly sure either where it is failing, thanks for spotting this.

File: app/views/includes/govuk_wrapper.scala.html (hunk 2 above)
Comment (mywyau) [reply app/views/includes/govuk_wrapper.scala.html:60]: think it should be internal header as it is an internal stride service, but yeah need to fix the url

File: app/views/partials/case_trader.scala.html [hunk 5]
```diff
@@ -22,8 +22,9 @@
 
 @tab_heading("trader", "Applicant")
 
+<h3 class="govuk-heading-m" id="trader-submitted-heading">@messages("case.application.holder.section_heading")</h3>
```
[the remark this answers is not in the document]
Comment (mywyau) [reply app/views/partials/case_trader.scala.html:25]: Ah yeah made it the same and was intentional when touching this page/component saw it was inconsistent.

File: app/views/includes/govuk_wrapper.scala.html (hunk 2 above)
Comment (mywyau) [reply app/views/includes/govuk_wrapper.scala.html:60]: okay just an update should now nav back to the dashboard

https://user-images.githubusercontent.com/47248104/205640044-cff24fbe-24d4-47fa-ad31-1a78d7e46efa.mov

File: app/views/includes/govuk_wrapper.scala.html (hunk 3 above)
Comment (mywyau) [reply app/views/includes/govuk_wrapper.scala.html:62]: see above vid

File: app/views/includes/govuk_wrapper.scala.html (hunk 4 above)
Comment (mywyau) [reply app/views/includes/govuk_wrapper.scala.html:64]: see above vid both nav back to dashboard
Screenshot 2022-12-05 at 12 44 29
Screenshot 2022-12-05 at 12 44 50

File: app/views/forms/components/repayment_input_checkbox.scala.html (hunk 1 above)
Comment (mywyau) [reply app/views/forms/components/repayment_input_checkbox.scala.html:31]: hopefully should remember the checked status now reverted back/similar to old code

File: app/views/forms/components/input_checkbox.scala.html [hunk 6]
```diff
@@ -28,25 +28,18 @@
...
-        @if(field.hasErrors){
-            aria-describedby="error-message-@{field.id}-input"
-        }
-        @if(field.value.contains(s"$value")){
-            checked="checked"
-        }
-        @if(data_target.isDefined) {
-            data-aria-controls="@data_target.get"
-        }
+            class="govuk-checkboxes__input @inputClass"
```
[the remark this answers is not in the document]
Comment (mywyau) [reply app/views/forms/components/input_checkbox.scala.html:33]: Screenshot 2022-12-05 at 14 25 43
Should retain error on date field

File: app/views/forms/components/input_checkbox.scala.html [hunk 7]
```diff
@@ -28,25 +28,18 @@
 
 <div class="govuk-form-group @if(field.hasErrors){govuk-form-group--error}">
 
-    <div class="govuk-checkboxes__item" @if(data_target.isDefined){data-target="@data_target.get"}>
+    <div class="govuk-checkboxes__item" data-target=@if(data_target.isDefined){"@data_target.get"}>
```
[the remark this answers is not in the document]
Comment (mywyau) [reply app/views/forms/components/input_checkbox.scala.html]: reverted to old version for now as it wasn't an ally issue. So it does not render invalid html if passed None. 
Can be looked at in the future if the service gets refactored.

File: app/views/forms/components/input_checkbox.scala.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (frikit) [reply app/views/forms/components/input_checkbox.scala.html:33]: fixed

File: app/views/includes/govuk_wrapper.scala.html (hunk 2 above)
Comment (frikit) [reply app/views/includes/govuk_wrapper.scala.html:60]: fixed

**Review by frikit — APPROVED** [review]

**Pull request merged**